### PR TITLE
IDEX-3736: allow setting custom docker containers host

### PR DESCRIPTION
**Pull request by garagatyi**

### CODE REVIEW

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineFactory.java [hunk 1]
```diff
@@ -39,4 +40,7 @@ Instance createInstance(@Assisted MachineState machineState,
 
     DockerNode createNode(@Assisted("workspace") String workspaceId,
                           @Assisted("container") String containerId) throws MachineException;
+
+    DockerInstanceMetadata createMetadata(@Assisted ContainerInfo containerInfo,
```
Comment (vzhukovskii) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineFactory.java]: Javadoc

Comment (vzhukovskii) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineFactory.java]: What if I'll pass null as parameters?

Comment (garagatyi) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineFactory.java]: Assisted injection doesn't allow pass null, Guice will throw exception.

**Comment by skabashnyuk** [comment]
ок

**Pull request merged**